### PR TITLE
Update CMake versions to test

### LINKED ISSUE #66: Test recent CMake versions
**Is your feature request related to a problem? Please describe.**
Unit tests are performed on quite outdated versions of CMake (see [this run](https://github.com/CMakePP/CMakePPLang/actions/runs/5004883911)).

**Describe the solution you'd like**
While we should definitely test on the oldest version of CMake that we plan to support, I think we should also be testing on the latest CMake version (v3.26.3 at the time of writing) since users are more likely to have that than v3.14.7.

The `lukka/get-cmake@latest` [action](https://github.com/marketplace/actions/get-cmake) is useful for this, since you can specify `latest` and `latestrc` in the CMake version matrix in the CI to always fetch the latest stable and release candidates of CMake, respectively. 

**Describe alternatives you've considered**
We could manually change the version matrix when new versions of CMake are released, but if we can automate it, I think we should.

**Additional Context**
The step needed to use the `lukka/get-cmake@latest` action would be:
```cmake
- name: Get latest CMake and Ninja
  uses: lukka/get-cmake@latest
  with:
    cmakeVersion: ${{ env.cmake_version }}
```

### PR DESCRIPTION
**Pull request by zachcran**
**Is this pull request associated with an issue(s)?**
Fixes #66.

**Description**
Update which CMake versions are tested during unit testing. This switches the way we fetch CMake to use `lukka/get-cmake@latest` ([link](https://github.com/marketplace/actions/get-cmake)).

**TODOs**
- [x] Make sure that we have covered all the versions we want to.
- [x] Make sure that the `latest` keyword works to get the latest stable CMake release.

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -8,23 +8,31 @@ jobs:
...
+#       - name: get cmake
+#         env:
+#           url_prefix: https://github.com/Kitware/CMake/releases/download
+#         run: |
+#           cmake_script=cmake-${cmake_version}-${os}.sh
+#           wget "${url_prefix}/v${cmake_version}/${cmake_script}"
+#           yes | /bin/sh "${cmake_script}"
+
+      - name: Get latest CMake and Ninja
+        uses: lukka/get-cmake@latest
```
Comment (ryanmrichard) [inline .github/workflows/unit_test.yml:27]: This gets ninja too!?!? This is a way better solution!!

**Comment by zachcran** [comment]
@ryanmrichard What versions of CMake do you think we should test? I only kept the oldest version (3.14.7) that we were testing before and added the latest stable release and latest release candidate.

**Comment by ryanmrichard** [comment]
I think the oldest, latest, and latestrc are good for now.

**Comment by zachcran** [comment]
This PR is r2g once the tests pass then. It will likely need @ryanmrichard to manually merge it, since it won't be passing the two required tests for CMake versions we no longer include.

**Comment by zachcran** [comment]
@ryanmrichard r2g

**Review by ryanmrichard — APPROVED** [review]

**Pull request merged**